### DISCUSSION TITLE
How to access additional Form Fields added to a Form Template?

### DISCUSSION THREAD

**Discussion by chrismlusk** [verbb/formie#1397] [Q&A] [answered]
When creating a form template, there is the "Form Fields" tab where we can add additional elements that are shown in tabs when building the form. How can I access these fields when rendering my forms? I can't seem to track down anything in the docs other than [an acknowledgement that you can add these fields](https://verbb.io/craft-plugins/formie/docs/feature-tour/form-templates#form-fields).

I am not using a custom template, FYI.

**Comment by engram-design** [comment, marked as answer]
You can access them like any other element. Sorry if that's assumed behaviour.

If you add a field with `myPlainText` as the handle, you can access that via `form.myPlainText`.

```
{% set form = craft.formie.forms({ handle: 'contactForm' }).one() %}

{{ form.myPlainText }}
```

**Reply by chrismlusk** [reply]
Of course 🤦🏼 

In my defense, the examples of field rendering I see in the docs are:

```twig
{# 1. Getting the custom fields #}
{% for field in form.getCustomFields() %}
    {{ craft.formie.renderField(form, field) }}
{% endfor %}

{# 2. Direct access by field handle #}
{{ craft.formie.renderField(form, 'myField') }}
```

And neither of these work.

The first only has Formie fields, which makes sense. The second returns null.

But, I see that `{{ form.myField }}` does in fact work. Thank you!

**Reply by engram-design** [reply]
Right, so `form.getCustomFields()` is _technically_ the Formie Form fields, not the Formie Form custom fields. I _know_ the function is `getCustomFields`, but that's a change made in Craft 4 with field layouts.

I blame having a confusing concept of Custom Craft fields and Formie Form fields built into forms for the confusion...
